src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+        system.assertEquals('Direct City Edit', mapAccIdAcc.values()[1].BillingCity);
+        system.assertEquals(true, mapAccIdAcc.values()[1].is_Address_Override__c);
+ 
+         // verify the previous addresses didn't get Default cleared.
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c from Address__c];
+        system.assertEquals(4, listAddr.size());
+        for (Address__c addr : listAddr) {
+            boolean isNewEdit = (addr.MailingStreet__c.contains('Direct Street Edit'));
+            if(isNewEdit) {
+                system.assertNotEquals(false, addr.Default_Address__c);
+            }
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Update Account's billing address with just a case change, to cause an update to default Address object  
+    verify:
+        existing Default Address updated
+        Account has updated Address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateAddrExistingCaseOnly() {
+        configSettings();
+        
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // modify some of the account addresses directly
+        // NOTE: we only modify 1 field, so it will be treated as an update to an existing address!
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = 'Direct Street Edit';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;
+
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingState = 'Washington';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;
+
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = ' direct street  edit ';  // whitespace and casing should not count as a real edit.
+            accsAddrs.accs[i].BillingState = 'Oregon';
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;
+
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            accsAddrs.accs[i].BillingStreet = ' DIRECT STREET  EDIT ';  // only change casing.
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update accsAddrs.accs;
+        Test.stopTest();
+        
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause or contain the number of records redurned, or adding a `LIMIT` clause to cap the maximum number of records returned.